### PR TITLE
Improvements to `spoom bump`

### LINKED ISSUE #10: Error parser bug when no Ruby file is provided
When `srb tc` is called on a directory without any Ruby files, the error output is:

```
"You must pass either `-e` or at least one folder or ruby file.\n\nTypechecker for Ruby\nUsage:\n  sorbet [OPTION...] <path 1> <path 2> ...\n\n  -e, string     Parse an inline ruby string (default: \"\")\n  -q, --quiet    Silence all non-critical errors\n  -v, --verbose  Verbosity level [0-3]\n  -h,            Show short help\n      --help     Show long help\n      --version  Show version\n\n"
```

This is parsed by `Spoom::Sorbet::Errors::Parser.parse_string` as:

```
[#<Spoom::Sorbet::Errors::Error:0x00007f926231a190
  @code=nil,
  @file="You must pass either `-e` or at least one folder or ruby file.\n",
  @line=nil,
  @message=nil,
  @more=[]>,
 #<Spoom::Sorbet::Errors::Error:0x00007f9241966a10 @code=nil, @file="Typechecker for Ruby\n", @line=nil, @message=nil, @more=[]>,
 #<Spoom::Sorbet::Errors::Error:0x00007f9242840518
  @code=nil,
  @file="Usage",
  @line=0,
  @message=nil,
  @more=
   ["  sorbet [OPTION...] <path 1> <path 2> ...\n",
    "  -e, string     Parse an inline ruby string (default: \"\")\n",
    "  -q, --quiet    Silence all non-critical errors\n",
    "  -v, --verbose  Verbosity level [0-3]\n",
    "  -h,            Show short help\n",
    "      --help     Show long help\n",
    "      --version  Show version\n"]>]
```

### PR DESCRIPTION
**Pull request by Morriar**
This PR brings some improvements to the `spoom bump` command:

* `spoom bump` uses `in_sorbet_project!` and displays an error if ran outside of a Sorbet project
* Some improvements to the error message when ran outside a Sorbet project
* `spoom bump` uses the `--path` option, so we can `spoom bump lib/ --path my/project`
* Avoid reverting files from other directories when bumping files with errors

Also add a few tests to show all the new behavior.

Fixes #10 and #50.

### CODE REVIEW

**Review by KaanOzkan — APPROVED** [review]

File: test/spoom/cli/config_test.rb [hunk 1]
```diff
@@ -17,10 +17,10 @@ def teardown
       end
 
       def test_return_error_if_no_sorbet_config
-        _, err, _ = @project.bundle_exec("spoom config")
-        assert_equal(<<~MSG, err)
-          Error: not in a Sorbet project (no sorbet/config)
-        MSG
+        out, err, status = @project.bundle_exec("spoom config --no-color")
+        assert_empty(out)
+        assert_equal("Error: not in a Sorbet project (sorbet/config not found)", err.split("\n").first)
```
Comment (paracycle) [inline test/spoom/cli/config_test.rb:22]: Again, recommend `err.lines.first.chomp` over `split` here.

File: test/spoom/cli/coverage_test.rb [hunk 2]
```diff
@@ -82,6 +82,14 @@ def test_display_metrics
         assert_equal(0, Dir.glob("#{@project.path}/spoom_data/*.json").size)
       end
 
+      def test_snapshot_outside_sorbet_dir
+        @project.remove("sorbet/config")
+        out, err, status = @project.bundle_exec("spoom coverage snapshot --no-color")
+        assert_empty(out)
+        assert_equal("Error: not in a Sorbet project (sorbet/config not found)", err.split("\n").first)
```
Comment (paracycle) [inline test/spoom/cli/coverage_test.rb:89]: Same

File: test/spoom/cli/coverage_test.rb [hunk 3]
```diff
@@ -156,12 +164,10 @@ def test_display_metrics_with_path_option
 
       def test_timeline_outside_sorbet_dir
         @project.remove("sorbet/config")
-        out, err, status = @project.bundle_exec("spoom coverage snapshot")
+        out, err, status = @project.bundle_exec("spoom coverage timeline --no-color")
+        assert_empty(out)
+        assert_equal("Error: not in a Sorbet project (sorbet/config not found)", err.split("\n").first)
```
Comment (paracycle) [inline test/spoom/cli/coverage_test.rb:169]: And same...

File: test/spoom/cli/coverage_test.rb [hunk 4]
```diff
@@ -343,6 +349,14 @@ def test_timeline_with_path_option
         project.destroy
       end
 
+      def test_report_outside_sorbet_dir
+        @project.remove("sorbet/config")
+        out, err, status = @project.bundle_exec("spoom coverage report --no-color")
+        assert_empty(out)
+        assert_equal("Error: not in a Sorbet project (sorbet/config not found)", err.split("\n").first)
```
Comment (paracycle) [inline test/spoom/cli/coverage_test.rb:356]: And same...

File: test/spoom/cli/lsp_test.rb [hunk 5]
```diff
@@ -21,10 +21,10 @@ def teardown
 
       def test_cant_open_without_config
         @project.remove("sorbet/config")
-        _, err = @project.bundle_exec("spoom lsp --no-color find Foo")
-        assert_equal(<<~MSG, err)
-          Error: not in a Sorbet project (no sorbet/config)
-        MSG
+        out, err, status = @project.bundle_exec("spoom lsp --no-color find Foo")
+        assert_empty(out)
+        assert_equal("Error: not in a Sorbet project (sorbet/config not found)", err.split("\n").first)
```
Comment (paracycle) [inline test/spoom/cli/lsp_test.rb:26]: Yet again

File: test/spoom/cli/run_test.rb [hunk 6]
```diff
@@ -31,12 +31,12 @@ def teardown
...
-      def test_return_error_if_no_sorbet_config
+      def test_timeline_outside_sorbet_dir
         @project.remove("sorbet/config")
-        _, err = @project.bundle_exec("spoom tc")
-        assert_equal(<<~MSG, err)
-          Error: not in a Sorbet project (no sorbet/config)
-        MSG
+        out, err, status = @project.bundle_exec("spoom tc --no-color")
+        assert_empty(out)
+        assert_equal("Error: not in a Sorbet project (sorbet/config not found)", err.split("\n").first)
```
Comment (paracycle) [inline test/spoom/cli/run_test.rb:38]: One last time...

**Review by paracycle — APPROVED** [review]

**Comment by Morriar** [comment]
Replaced all `split` by `lines.first.chomp`

**Pull request merged**